src/reconcile/rec_visibility.c

```diff
@@ -758,3 +758,253 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
+    upd_select->upd = NULL;
+    upd_select->upd_saved = false;
+
+    page = r->page;
+    upd = NULL;
+    upd_memsize = 0;
+    max_ts = WT_TS_NONE;
+    max_txn = WT_TXN_NONE;
+    list_prepared = list_uncommitted = false;
+
+    /*
+     * If called with a WT_INSERT item, use its WT_UPDATE list (which must
+     * exist), otherwise check for an on-page row-store WT_UPDATE list
+     * (which may not exist). Return immediately if the item has no updates.
+     */
+    if (ins != NULL)
+        first_upd = ins->upd;
+    else if ((first_upd = WT_ROW_UPDATE(page, ripcip)) == NULL)
+        return (0);
+
+    for (upd = first_upd; upd != NULL; upd = upd->next) {
+        if ((txnid = upd->txnid) == WT_TXN_ABORTED)
+            continue;
+
+        ++r->updates_seen;
+        upd_memsize += WT_UPDATE_MEMSIZE(upd);
+
+        /* Track the max txnid */
+        if (WT_TXNID_LT(max_txn, txnid))
+            max_txn = txnid;
+
+        /*
+         * Check whether the update was committed before reconciliation started. The global commit
+         * point can move forward during reconciliation so we use a cached copy to avoid races when
+         * a concurrent transaction commits or rolls back while we are examining its updates. As
+         * prepared transaction IDs are globally visible, need to check the update state as well.
+         */
+        if (F_ISSET(r, WT_REC_VISIBLE_ALL) ? WT_TXNID_LE(r->last_running, txnid) :
+                                             !__txn_visible_id(session, txnid)) {
+            r->update_uncommitted = list_uncommitted = true;
+            continue;
+        }
+        if (upd->prepare_state == WT_PREPARE_LOCKED ||
+          upd->prepare_state == WT_PREPARE_INPROGRESS) {
+            list_prepared = true;
+            if (upd->start_ts > max_ts)
+                max_ts = upd->start_ts;
+            continue;
+        }
+
+        /* Track the first update with non-zero timestamp. */
+        if (upd->durable_ts > max_ts)
+            max_ts = upd->durable_ts;
+
+        /*
+         * Always select the newest committed update to write to disk
+         */
+        if (upd_select->upd == NULL)
+            upd_select->upd = upd;
+
+        if (!__rec_update_durable(session, r, upd)) {
+            if (F_ISSET(r, WT_REC_EVICT))
+                ++r->updates_unstable;
+
+            /*
+             * Rare case: when applications run at low isolation levels, update/restore eviction may
+             * see a stable update followed by an uncommitted update. Give up in that case: we need
+             * to discard updates from the stable update and older for correctness and we can't
+             * discard an uncommitted update.
+             */
+            if (F_ISSET(r, WT_REC_UPDATE_RESTORE) && upd_select->upd != NULL &&
+              (list_prepared || list_uncommitted))
+                return (__wt_set_return(session, EBUSY));
+
+            continue;
+        }
+
+        if (!F_ISSET(r, WT_REC_EVICT))
```

Comment:
If we don't need to track all the counters, we can remove this check as well and just break.